qiskit/quantum_info/states/statevector.py

```diff
@@ -113,6 +113,30 @@ def __init__(
                 raise QiskitError("Invalid input: not a vector or column-vector.")
         super().__init__(op_shape=OpShape.auto(shape=shape, dims_l=dims, num_qubits_r=0))
 
+    @classmethod
+    def from_circuit(cls, circuit, input_state=None):
```

Comment:
Did you want to mirror the options from `Operator.from_layout` for ignoring the layout?

I don't think we want the `initial_layout` or `final_layout` options as personally I think those were mistakes I made in that interface, but the ignore option seems like it would be useful here.